statistics/handle/handle.go

```diff
@@ -251,28 +267,47 @@ func (h *Handle) updateStatsCache(newCache statsCache) {
 	h.statsCache.Lock()
 	oldCache := h.statsCache.Load().(statsCache)
 	if oldCache.version <= newCache.version {
+		h.statsCache.memTracker.Consume(newCache.memUsage - oldCache.memUsage)
 		h.statsCache.Store(newCache)
 	}
 	h.statsCache.Unlock()
 }
 
 func (sc statsCache) copy() statsCache {
-	newCache := statsCache{tables: make(map[int64]*statistics.Table, len(sc.tables)), version: sc.version}
+	newCache := statsCache{tables: make(map[int64]*statistics.Table, len(sc.tables)),
+		version:  sc.version,
+		memUsage: sc.memUsage}
 	for k, v := range sc.tables {
 		newCache.tables[k] = v
 	}
 	return newCache
 }
 
+func (sc statsCache) initMemoryUsage() {
+	sum := int64(0)
+	for _, tb := range sc.tables {
+		sum += tb.MemoryUsage()
+	}
+	sc.memUsage = sum
```

Comment:
`sc.memUsage` should be updable? when some tables/columns/indexes are deleted or added.

Reply:
`initMemoryUsage` only invoked in handle.InitStats().
`sc.memUsage`will be update in copied stateCache `newCache`  by invoke statsCache.Upate() , and `newCache`  will be stored in handle.StateCache in handle.updateStatsCache().